tests/api/test_api.py

```diff
@@ -695,3 +703,105 @@ async def get_collection(
+        postgres_dbname=database.dbname,
+        testing=True,
+        use_api_hydrate=hydrate,
+        enable_response_models=validation,
+    )
+    extensions = []
+    post_request_model = create_post_request_model(extensions, base_model=PgstacSearch)
+    api = StacApi(
+        client=CoreCrudClient(post_request_model=post_request_model),
+        settings=settings,
+        extensions=extensions,
+        search_post_request_model=post_request_model,
+    )
+    app = api.app
+    await connect_to_db(app)
+    try:
+        async with AsyncClient(transport=ASGITransport(app=app)) as client:
+            landing = await client.get("http://test/")
+            assert landing.status_code == 200, landing.text
+
+            collection = await client.get("http://test/collections/test-collection")
+            assert collection.status_code == 200, collection.text
+
+            collections = await client.get("http://test/collections")
+            assert collections.status_code == 200, collections.text
+
+            item = await client.get(
+                "http://test/collections/test-collection/items/test-item"
+            )
+            assert item.status_code == 200, item.text
+
+            item_collection = await client.get(
+                "http://test/collections/test-collection/items",
+                params={"limit": 10},
+            )
+            assert item_collection.status_code == 200, item_collection.text
+
+            get_search = await client.get(
+                "http://test/search",
+                params={
+                    "collections": ["test-collection"],
+                },
+            )
+            assert get_search.status_code == 200, get_search.text
+
+            post_search = await client.post(
+                "http://test/search",
+                json={
+                    "collections": ["test-collection"],
+                },
+            )
+            assert post_search.status_code == 200, post_search.text
+
+            get_search = await client.get(
+                "http://test/search",
+                params={
+                    "collections": ["test-collection"],
+                    "fields": "properties.datetime",
+                },
+            )
+            # fields should be ignored
+            assert get_search.status_code == 200, get_search.text
+            props = get_search.json()["features"][0]["properties"]
+            assert len(props) > 1
+
+            post_search = await client.post(
+                "http://test/search",
+                json={
+                    "collections": ["test-collection"],
+                    "fields": {
+                        "include": ["properties.datetime"],
+                    },
+                },
+            )
+            # fields should be ignored
+            assert post_search.status_code == 200, post_search.text
+            props = get_search.json()["features"][0]["properties"]
+            assert len(props) > 1
```

Comment:
IMO we could also change the behaviour to have fastapi error when passing an unwanted body parameter 

This could be done by changing the pydantic model configuration `model_config={"extra"='forbid'}